pkg/client/private_cloud_gateway.go

```diff
@@ -20,25 +21,48 @@ func (h *V1Client) CreateIpPool(pcgUID string, pool *models.V1IPPoolInputEntity)
 	}
 }
 
+// get ip pool by uid
 func (h *V1Client) GetIpPool(pcgUID, poolUID string) (*models.V1IPPoolEntity, error) {
+	pools, err := h.GetIpPools(pcgUID)
+	if err != nil {
+		return nil, err
+	}
+	for _, pool := range pools {
+		if pool.Metadata.UID == poolUID {
+			return pool, nil
+		}
+	}
+	return nil, nil
```

Comment:
should be returning an error here, but you aren't

Reply:
thank you for checking, actually already fixed it locally but still verifying on the env.